### ISSUE TITLE
Node.js is adopting an official Discord server for the Community - CoC review is needed

### ISSUE THREAD

**Issue by ovflowd** [openjs-foundation/cross-project-council#1347]
Hey folks 👋 the Node.js project has been in talks about creating an official Discord community by repurposing an existing Discord community (details here: https://github.com/nodejs/admin/issues/872);

This issue requests that the Foundation review its CoC to verify that it follows its requirements. Since it will become an official space for an official OpenJS project, it is essential that their CoC fulfills our requirements.

Their current CoC also follows the Contributor Covenant 2.0 (based on Node.js's CoC but adapted for use on the Discord platform; since Discord is a communication app/platform, it is very different from GitHub.

Here is the diff: https://github.com/nodeiflux/documents/commit/a976e0836d337d53c8d64be9e9e7f42de6983675#diff-13936941ef00c85a7f555aa9ec176552f9965f0d2eeadd5ea95c541d5cd09bfb

I've requested @vcarl (the owner of said Discord community) to, if possible:

- Update the report email address to `report@nodejs.org`
- Update the base version of their Contributor Covenant to Contributor Covenant 2.1

I'd also like the CPC to review their CoC to ensure everything is good. Thank you!

**Comment by ljharb** [comment]
why are the changes needed? both the node slack and IRC have used node's code of conduct verbatim for many years without any issue.

**Comment by ovflowd** [comment]
> why are the changes needed? both the node slack and IRC have used node's code of conduct verbatim for many years without any issue.

They are already using this CoC -- I want to retify that this CoC is also OK for us. It has also been working fine for them for a long period of years.

**Comment by mhdawson** [comment]
@ovflowd my understanding was that the existing moderation team was going to continue to moderate the Discord server with the Node.js moderation team being added for visibility.

In that context would it make sense to have a different report email alias (maybe report-discord@nodejs.org) that would include the discord moderation team members as well as the Node.js moderation team ?

**Comment by ovflowd** [comment]
> @ovflowd my understanding was that the existing moderation team was going to continue to moderate the Discord server with the Node.js moderation team being added for visibility.
> 
> 
> 
> In that context would it make sense to have a different report email alias (maybe report-discord@nodejs.org) that would include the discord moderation team members as well as the Node.js moderation team ?

Yup, that's the idea. I already opened a PR on nodejs/email for that, if you'd like to chek that 👀

**Comment by PaulaPaul** [comment]
No concerns - (this was discussed at the CPC call on 20 Aug) just wanted to share that it may be helpful to include the new Discord on the Node.js website and potentially the OpenJS Foundation website to help connet with the community.

**Comment by tobie** [comment]
Discussed in the CPC today. Waiting for follow-up from @ovflowd on the pending questions raised by the CPC in our last call: 

- What is the rationale for not using the foundation's CoC?
- How official will this server be? If this is a verified server it implies that it is tied to the foundation and hence will need to adopt the foundation's CoC verbatim.

**Comment by vcarl** [comment]
> What is the rationale for not using the foundation's CoC?

We are an independent moderation team that has demonstrated an ability to safely operate an online community space since 2018, we have not been asked to substantively alter our moderation processes. I found that the existing Node.js Code of Conduct was mostly in line with how we've been operating, but felt that our own CoC was important to distinguish that we are independently operated (though ultimately accountable to the foundation). 

[The changes I've made](https://github.com/nodeiflux/documents/commit/a976e0836d337d53c8d64be9e9e7f42de6983675#diff-13936941ef00c85a7f555aa9ec176552f9965f0d2eeadd5ea95c541d5cd09bfb) are:
- changed the report email (I need to update this, this change sparked a conversation about what email to use)
- elaborated on expectations for changes to behavior requested by moderators over time.
- changed "temporary ban" to "timeout", as that's the name of a Discord feature for limiting participation
- changed 'serious violation' to 'violation' in describing when a timeout may be used
- changed 'permanent ban' to 'ban' in the Discord, which reflects the process we've used for appealing a ban historically
- added details about who is eligible to appeal a ban
- credited Node.js and Reactiflux as inspiration for the CoC

**Comment by ljharb** [comment]
@vcarl "permanent" doesn't mean "can't be appealed", it means "has no automatic expiry"

**Comment by ovflowd** [comment]
> No concerns - (this was discussed at the CPC call on 20 Aug) just wanted to share that it may be helpful to include the new Discord on the Node.js website and potentially the OpenJS Foundation website to help connet with the community.

It will be done at the moment we release it!

**Comment by PaulaPaul** [comment]
@joesepi will reach out to @ovflowd to chat (Paula to sit in for education/help if appropriate)

**Comment by tobie** [comment]
CPC call update: discussed this today. Want to find a pathway forward. @joesepi to follow-up with @ovflowd.

**Comment by ovflowd** [comment]
CTA:

- Check-in with Carl regarding the status of the work on his side :)

**Comment by tobie** [comment]
**Notes from today's CPC call:**

Waiting on @ovflowd driving the conversation on their end. Removing CPC label for now. @ovflowd, feel free to add the label back in once you're ready for the CPC's input.

**Comment by ovflowd** [comment]
@mcollina happy to mention that they adopted our Coc - https://github.com/nodeiflux/documents/blob/main/CONDUCT.md

This doesn't change the already approved stance the Node.js TSC had before. What are our next steps here?

I recall the CPC needs to approve the usage of the Node.js brand on the Discord server, right? Or what else needs to be approved? :)

**Comment by mcollina** [comment]
The next steps are:

1. agreement with the TSC that nodeiflux becomes the official discord
2. adding the key personnel of nodeiflux to the Node.js org for official comunications
3. adding key personnel to Node.js to nodeiflux as admins
4. optionally rebrand nodeiflux as "Node.js"

I don't think the CPC should be involved at all.

**Comment by ovflowd** [comment]
> The next steps are:
> 
> 1. agreement with the TSC that nodeiflux becomes the official discord
> 2. adding the key personnel of nodeiflux to the Node.js org for official comunications
> 3. adding key personnel to Node.js to nodeiflux as admins
> 4. optionally rebrand nodeiflux as "Node.js"
> 
> I don't think the CPC should be involved at all.

1. This was already done within the issue opened within the TSC
2. That is already done. They already have an alias email and a direct Slack channel to communicate with us (Node.js Moderation team)
3. To be done, once the server is almost ready
4. That is definitely going to happen; The whole point is to be named "Node.js" and contain its branding 🙈

**Comment by ovflowd** [comment]
Also here's a copy of their CoC now: https://github.com/nodeiflux/documents/blob/main/CONDUCT.md

**Comment by mcollina** [comment]
> This was already done within the issue opened within the TSC

Frankly, this has not happened. The TSC has approved it, but up to the latest call with the CPC, there was no clarity of what it meant to be official. So yeah, I think it's better to have another pass, with the lead of nodeiflux in the call.

**Comment by ovflowd** [comment]
> > This was already done within the issue opened within the TSC
> 
> Frankly, this has not happened. The TSC has approved it, but up to the latest call with the CPC, there was no clarity of what it meant to be official. So yeah, I think it's better to have another pass, with the lead of nodeiflux in the call.

Is Carl's presence required? Since I'm the intermediary, I can be on the TSC call. Otherwise, @vcarl, let us know if you want to join the TSC call, and we can schedule it.

**Comment by mcollina** [comment]
We would need @vcarl.

**Comment by vcarl** [comment]
Happy to join, not sure where to find details of how/when!

**Comment by ovflowd** [comment]
@mhdawson, which TSC meeting would you suggest?

**Comment by ovflowd** [comment]
And @vcarl, you can see it here: https://nodejs.org/calendar when the TSC meetings happen. Usually, every Wednesday at a fixed time. Would that time work for you?

We can then get from @mhdawson which meeting we should join.

**Comment by mhdawson** [comment]
This week at 10 ET on the 23rd would work. Next week as well although I'll be looking for a Chair for that meeting since I'm  going to miss it. The week after is NodeConf.eu so I'm going to suggest we skip that one.  

@vcarl, @ovflowd does this week work for you two?

**Comment by tobie** [comment]
**Notes from this week's CPC call:**

  * This has been punted back to Node.js TSC
  - [ ] Open issue to discuss a foundation-wide discord (like we have a foundation-wide slack)

**Comment by ovflowd** [comment]
Update from Node.js's TSC: The TSC has approved and decided to move forward with the adoption of this server as the official Node.js Discord server. cc @mhdawson for more details

**Comment by mhdawson** [comment]
@ovflowd I think this issue could be closed as it was agreed that the Discord server would adopt the CoC of the foundation right?

**Comment by vcarl** [comment]
We did agree that and it has been changed! Currently forwards via link in [CONDUCT.md](https://github.com/nodeiflux/documents/blob/main/CONDUCT.md) in nodeiflux/documents
